### PR TITLE
Added new settings view and main to pagespeed insights datastore.

### PR DESCRIPTION
**Pull request by ShahAaron**
## Summary

Addresses issue #1501 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/pagespeed-insights/index.js [hunk 1]
```diff
@@ -20,3 +20,17 @@
...
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Add components to the settings page.
+ */
+addFilter( 'googlesitekit.ModuleSettingsDetails-pagespeed-insights',
+	'googlesitekit.PageSpeedInsightsModuleSettingsDetails',
+	fillFilterWithComponent( PageSpeedInsightsSettings ), 10 );
```
Comment (aaemnnosttv) [inline assets/js/modules/pagespeed-insights/index.js:36]: Formatting, for consistency, also removing the default priority from `addFilter` as we aren't using this in any of the other new components. Let's only add it if we need it.

```suggestion
addFilter(
	'googlesitekit.ModuleSettingsDetails-pagespeed-insights',
	'googlesitekit.PageSpeedInsightsModuleSettingsDetails',
	fillFilterWithComponent( PageSpeedInsightsSettings )
);
```

File: stories/module-pagespeed-insights-settings.stories.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import { SettingsMain as PageSpeedInsightsSettings } from '../assets/js/modules/pagespeed-insights/settings';
+import { fillFilterWithComponent } from '../assets/js/util';
+import { WithTestRegistry } from '../tests/js/utils';
+
+function filterPageSpeedInsightsSettings() {
+	removeAllFilters( 'googlesitekit.ModuleSettingsDetails-pagespeed-insights' );
+	addFilter(
+		'googlesitekit.ModuleSettingsDetails-pagespeed-insights',
+		'googlesitekit.PageSpeedInsightsModuleSettingsDetails',
+		fillFilterWithComponent( PageSpeedInsightsSettings ), 10 );
```
Comment (aaemnnosttv) [inline stories/module-pagespeed-insights-settings.stories.js:42]:
```suggestion
		fillFilterWithComponent( PageSpeedInsightsSettings )
 );
```

File: assets/js/modules/pagespeed-insights/index.js (hunk 1 above)
Comment (ShahAaron) [reply assets/js/modules/pagespeed-insights/index.js]: Updated

File: stories/module-pagespeed-insights-settings.stories.js (hunk 2 above)
Comment (ShahAaron) [reply stories/module-pagespeed-insights-settings.stories.js]: Updated

**Pull request merged**